services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/sts/StsManagerImplAutowiredTest.java

```diff
@@ -0,0 +1,298 @@
+	private static final String SYNAPSE_BUCKET = CONFIG.getS3Bucket();
+
+	@Autowired
+	private AuthenticationManager authManager;
+
+	@Autowired
+	private EntityManager entityManager;
+
+	@Autowired
+	private FileHandleManager fileHandleManager;
+
+	@Autowired
+	private MultipartManager multipartManager;
+
+	@Autowired
+	private ProjectSettingsManager projectSettingsManager;
+
+	@Autowired
+	private SynapseS3Client s3Client;
+
+	@Autowired
+	private StsManager stsManager;
+
+	@Autowired
+	public UserManager userManager;
+
+	private List<FileHandle> fileHandlesToDelete;
+	private List<File> filesToDelete;
+	private String folderId;
+	private String projectId;
+	private UserInfo userInfo;
+	private String username;
+
+	@BeforeEach
+	public void beforeEach() {
+		// Create test user.
+		NewUser user = new NewUser();
+		username = UUID.randomUUID().toString();
+		user.setEmail(username + "@test.com");
+		user.setUserName(username);
+		userInfo = userManager.getUserInfo(userManager.createUser(user));
+		userInfo.getGroups().add(AuthorizationConstants.BOOTSTRAP_PRINCIPAL.CERTIFIED_USERS.getPrincipalId());
+
+		// User must agree to terms of use to get download privileges (and hence access the STS API).
+		authManager.setTermsOfUseAcceptance(userInfo.getId(), true);
+
+		// Create a test project which we will need.
+		Project project = new Project();
+		projectId = entityManager.createEntity(userInfo, project, null);
+
+		// Create folder, which is required for STS.
+		Folder folder = new Folder();
+		folder.setParentId(projectId);
+		folderId = entityManager.createEntity(userInfo, folder, null);
+
+		// Initialize lists of things to delete.
+		fileHandlesToDelete = new ArrayList<>();
+		filesToDelete = new ArrayList<>();
+	}
+
+	@AfterEach
+	public void afterEach() {
+		// Delete project.
+		//noinspection deprecation
+		entityManager.deleteEntity(userInfo, projectId);
+
+		// Delete file handles.
+		for (FileHandle fileHandle : fileHandlesToDelete) {
+			fileHandleManager.deleteFileHandle(userInfo, fileHandle.getId());
+		}
+
+		// Delete local files.
+		for (File file : filesToDelete) {
+			file.delete();
+		}
+
+		// Delete test user.
+		UserInfo adminUserInfo = userManager.getUserInfo(AuthorizationConstants.BOOTSTRAP_PRINCIPAL.THE_ADMIN_USER
+				.getPrincipalId());
+		userManager.deletePrincipal(adminUserInfo, Long.parseLong(userInfo.getId().toString()));
+	}
+
+	@Test
+	public void externalS3() throws Exception {
+		// Setup - In our test folder, create an inner folder. STS will be configured on the inner folder. STS
+		// credentials will be able to access files in the inner folder, but not the outer folder.
+		String outerFolderPath = "integration-test/" + "StsManagerImplAutowiredTest-" + UUID.randomUUID().toString();
+		String innerFolderPath = outerFolderPath + "/storage-location-root";
+		uploadFileToS3(outerFolderPath, "inaccessible.txt", "Dummy file content");
+		uploadFileToS3(innerFolderPath, "owner.txt", username);
```

Comment:
From code review discussion, we wanted to make sure we included the following test cases:

1. inaccessible.txt at the root of the S3 bucket
2. baseKey/owner.txt
3. baseKey/subfolder/file.txt

Should test both listing the folder and getting the file.

Also, we don't need a second level of folders for the baseKey.